emissary-core/src/destination/session/mod.rs

```diff
@@ -656,7 +664,42 @@ impl<R: Runtime> SessionManager<R> {
                     SessionError::Malformed
                 })?;
 
-                // TODO: verify `DateTime`
+                // locate `DateTime` message from the clove set
+                let Some(GarlicMessageBlock::DateTime { timestamp }) = clove_set
+                    .blocks
+                    .iter()
+                    .find(|clove| core::matches!(clove, GarlicMessageBlock::DateTime { .. }))
+                else {
+                    tracing::warn!(
+                        target: LOG_TARGET,
+                        id = %self.destination_id,
+                        "clove set doesn't contain `DateTime`",
+                    );
+
+                    return Err(SessionError::Malformed);
+                };
+
+                if clove_set
+                    .blocks
+                    .iter()
+                    .filter(|clove| core::matches!(clove, GarlicMessageBlock::DateTime { .. }))
+                    .count()
+                    > 1
+                {
+                    // Only one DateTime block is allowed
+                    return Err(SessionError::Malformed);
+                }
+
+                // The spec states that the timestamp will
+                // wrap for the year 2038 problem
+                let now = R::time_since_epoch();
+                let epoch_seconds = now.as_secs() as u32;
```

Comment:
```suggestion
                // The spec states that the timestamp will
                // wrap for the year 2038 problem
                let epoch_seconds = R::time_since_epoch().as_secs() as u32;
```

Where in the spec does it mention 2038?

Reply:
The 2038 is the general name of the problem as I know it; the 32bit timestamp roll over. For signed 32bit words its 2038, for unsigned its 2106. In the spec it says
```
timestamp :: Unix timestamp, unsigned seconds.
             Wraps around in 2106
```
I can reword the comment to make it more clear

Reply:
I think the comment can be removed